circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/HdfsGsCredentialDirectoryFactory.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.gcp;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import org.apache.hadoop.fs.Path;
+
+import com.hotels.bdp.circustrain.gcp.context.GCPSecurity;
+
+class HdfsGsCredentialDirectoryFactory {
+
+  static final String DEFAULT_HDFS_PREFIX = "hdfs:/tmp/ct-gcp-";
+  private String hdfsBaseDirectory;
+
+  Path newInstance(GCPSecurity security, String randomString) {
```

Comment:
I think you still might have issues here.
The default is fine, but you're changing the working dir.
Example:
```security.getDistributedFileSystemWorkingDirectory() == /home/user/```

Now you're code is going to change that to ```/home/user-random/```
Which means you need to have write permissions in `/home/` which you probably don't.

Make a random subdir it will solve everything.

`return new Path(hdfsBaseDirectory, randomString);`
Notice the ',' instead of +

This directory will be save to delete as well.